pkg/observer/controller.go

```diff
@@ -309,6 +370,15 @@ func (c *controller) Run() {
 	klog.V(3).Infof("waiting for %v to allow changes to settle", c.WaitInterval)
 	select {
 	case <-time.After(c.WaitInterval):
+		timeout := 0
+		retries := 20
+		for !c.safeToStartCycle() {
+			time.Sleep(10 * time.Second)
```

Comment:
not a huge fan of time.Sleep on the main thread. This will block the context closing.

Suggest break out into own function and using a select to wait on the both the time and the context.
```
for ...
    select {
	case <-time.After(c.WaitInterval):
          // if timeout >= retries { ...
        case <-c.stopCh:
           // program wants to exi nowt. return control, don't cycle